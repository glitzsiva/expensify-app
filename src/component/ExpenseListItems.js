import React from 'react';
import {Link} from 'react-router-dom';

const ExpenseListItems =({description, id,amount,createdAt})=>(
    <div>

        <Link to={`/edit/${id}`}>
            <h1>{description}</h1>
        </Link>
        <h5>{amount}-{createdAt}</h5>
        
    </div>
)
export default ExpenseListItems;