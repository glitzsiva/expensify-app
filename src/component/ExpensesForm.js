import React from 'react';
import moment from 'moment';
import {SingleDatePicker} from 'react-dates';



export default class ExpensesForm extends React.Component{
    
    constructor(props){
        super(props);

        this.state={
            description:props.expense? props.expense.description:'',
            note:props.expense? props.expense.note:'',
            amount:props.expense? props.expense.amount.toString():'',
            createdAt:props.expense? moment(props.expense.createdAt):moment(),
            CalendarFocused:false,
            error:''
    
        }
    }
    // state={
    //     description:props.expense ? props.description:'',
    // }


    changeDescription=(e)=>{
        const description=e.target.value;
        this.setState(()=>({description}));
    }
    changeNote=(e)=>{
        const note=e.target.value;
        this.setState(()=>({note}));
    }
    changeAmount=(e)=>{
        const amount=e.target.value;
        if(!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)){
            this.setState(()=>({amount}));
        }
        
    }
    onDateChange=(createdAt)=>{
        if(createdAt){
            this.setState(()=>({createdAt}))
        }
    }
    onFocusChange=()=>{
        this.setState(()=>({CalendarFocused:true}))
    }
    onSubmitExpenses=(e)=>{
        e.preventDefault();
        if(!this.state.description || !this.state.amount){
            this.setState(()=>({error:'please provide details'}))           
        }
        else{
            this.setState(()=>({error:''}))
            this.props.onSubmitCreate({
                description:this.state.description,
                amount:parseFloat(this.state.amount),
                note:this.state.note,
                createdAt:this.state.createdAt.valueOf()
            })
        }
    }

    render(){
        return (
            <div>

                {this.state.error && <h1>{this.state.error}</h1>}
                <form onSubmit={this.onSubmitExpenses}>
                    <input 
                        type='text'  
                        placeholder='Description' 
                        autoFocus 
                        value={this.state.description}
                        onChange={this.changeDescription}
                    />
                    <input 
                        type='number' 
                        placeholder='Amount' 
                        value={this.state.amount}
                        onChange={this.changeAmount}
                    />
                    <SingleDatePicker
                        date={this.state.createdAt} 
                        onDateChange={this.onDateChange} 
                        focused={this.state.CalendarFocused}
                        onFocusChange={this.onFocusChange}
                        isOutsideRange={()=> false}
                        numberOfMonths={1}
                    />

                    <textarea 
                        placeholder='add notes about your (Optional)'
                        value={this.state.note}
                        onChange={this.changeNote}
                    >
                    </textarea>
                    <button> Add Expenses</button>
                </form>
            </div>           
        )
    }
}