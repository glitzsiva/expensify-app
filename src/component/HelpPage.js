import React from 'react';
//import { NavLink} from 'react-router-dom';


const HelpPage=()=>(
    <div>
        helpPage
    </div>
);

export default HelpPage;