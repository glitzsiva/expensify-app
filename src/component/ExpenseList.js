import React from 'react';
import {connect} from 'react-redux';
import ExpenseListItems from './ExpenseListItems';
import visibleExpensesFilters from '../selectors/expenses';

export const ExpenseList =(props)=>(
    <div>
        <h1>Expense List</h1>
        {
            props.expenses.length >0 &&
            props.expenses.map((expense)=> <ExpenseListItems key={expense.id} {...expense} />
        )}
        
    </div>
);

const mapStateToProps =(state)=>{
    return {
        expenses:visibleExpensesFilters(state.expenses,state.filters)
    }
}

export default connect(mapStateToProps)(ExpenseList);