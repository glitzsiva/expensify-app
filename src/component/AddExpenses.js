import React from 'react';
import {connect} from 'react-redux';
import ExpensesForm from './ExpensesForm';
import {addExpenses} from '../actions/expenses'



const AddExpense=(props)=>(
    <div> 
        add Expenses page
        <ExpensesForm 
            onSubmitCreate={(expense)=>{
                    props.dispatch(addExpenses(expense));
                    props.history.push('/');
                }
            }
            
        />
    </div>
);
 export default connect()(AddExpense);