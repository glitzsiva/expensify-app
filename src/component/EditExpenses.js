import React from 'react';
import {connect} from 'react-redux';
import { editExpenses, removeExpenses } from '../actions/expenses';
import ExpensesForm from './ExpensesForm';


const EditExpenses=(props)=>{

    return(
        <div>
            <p>editExpenses{props.match.params.id}</p>
            <ExpensesForm                 
                expense={props.expense}
                onSubmitCreate={
                    (expense)=>{             // props.expense.id also access for edit expenses
                        props.dispatch(editExpenses(props.match.params.id,expense));
                        props.history.push('/')
                    }
                }
            />
            <button onClick={
                ()=>{
                    props.dispatch(
                        removeExpenses(
                            {id:props.match.params.id} 
                        )
                    )
                    props.history.push('/')
            }}>
                Remove
            </button>
        </div>
);

}
   
const mapStateToProps=(state,props)=>({expense:state.expenses.find((expense) => expense.id === props.match.params.id )})
export default connect(mapStateToProps)(EditExpenses);