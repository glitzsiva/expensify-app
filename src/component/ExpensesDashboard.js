import React from 'react';
import ExpenseList from './ExpenseList';
import ExpensesListFilters from './ExpensesListFilters';



const ExpenseDashboard=()=>(
    <div> 
        Dasboard page
        <ExpensesListFilters />
        <ExpenseList />
        
    </div>
);


export default ExpenseDashboard;