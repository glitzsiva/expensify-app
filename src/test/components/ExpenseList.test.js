import React from 'react';
import {shallow} from 'enzyme';
import {ExpenseList} from '../../component/ExpenseList';
import expense from '../fixtures/expenses'


test('Expenses List',()=>{
    const wrapper = shallow(<ExpenseList expenses={expense}/>);
    expect(wrapper.find('h1').text()).toEqual('Expense List')
    expect(wrapper).toMatchSnapshot();
})