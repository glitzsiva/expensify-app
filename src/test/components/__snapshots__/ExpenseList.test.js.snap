// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expenses List 1`] = `
<div>
  <h1>
    Expense List
  </h1>
  <ExpenseListItems
    amount={1000}
    createdAt={0}
    description="rent"
    id="1"
    key="1"
    note="rent amount"
  />
  <ExpenseListItems
    amount={10000}
    createdAt={345600000}
    description="rents"
    id="12"
    key="12"
    note="rents amounts"
  />
  <ExpenseListItems
    amount={100000}
    createdAt={-345600000}
    description="bill"
    id="123"
    key="123"
    note="bill amounts"
  />
</div>
`;
